version: '3.7'

services:
  web:
    image: reviewboard/rbserver:latest
    ports:
      - "5000:8000" # HTTP on the host
    volumes:
      - ./reviewboard/conf.py:/etc/reviewboard/conf.py
      - ./reviewboard/rc:/etc/reviewboard/rc.d
      - reviewboard-media:/code/rbserver/staticfiles
      - reviewboard-revborwse-cache:/code/rbserver/revborwse_cache
    env_file:
      - .env
    depends_on:
      - postgres
    command: rbserve --config /etc/reviewboard/conf.py

  postgres:
    image: postgres:latest
    volumes:
      - reviewboard-postgres-data:/var/lib/postgresql/data
    env_file:
      - .env
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U rbuser"]
      timeout: 10s
      retries: 30

  lighthouse:
    image: reviewboard/rbtests.lighthouse:latest
    volumes:
      - ./reviewboard/rc:/etc/reviewboard/rc.d
      - lighthouse-storage:/code/rbtests.lighthouse/storage
    env_file:
      - .env
    depends_on:
      - web
      - postgres
    command: rbtests.lighthouse --config /etc/reviewboard/rc.d/localrc.py --server http://web:8000

volumes:
  reviewboard-media:
  reviewboard-postgres-data:
  lighthouse-storage:

networks:
  default:
    driver: bridge